src/schemas/xspec.rnc

```diff
@@ -20,7 +20,7 @@ description =
 		attribute version { xs:NMTOKEN }?,		
 		attribute preserve-space { xs:NMTOKENS }?,
 		attribute schematron { xs:anyURI }?,
-		(global-param | import | scenario | pending)+ }
+		(global-param | import | scenario | pending | variable)+ }
 
```

Comment:
Is this `variable` necessary? Does this `variable` realize a use case which `global-param` can't do?

Reply:
If you're not using common attributes or `@as`, then as children of `description`, `variable` and `param` seem to produce the same result. I am not sure why the `global-param` pattern in the schema does not include common attributes and `@as`.

Aside from the question of supported attributes, someone might choose to use `param` for overrides of globals defined in the file to be tested, reserving `variable` for variables that are defined and used only in the XSpec test. Do you think it's worthwhile to support that distinction in the mental model?

Your question made me think about the namespace for variables that are defined and used in the XSpec test only. Maybe the test author should put them in the XSpec namespace as a matter of convention. In that convention, the namespace indicates which globals are overrides of the file being tested and which globals are in the test only, even if both kinds use the same element, `param`. Alternatively, maybe the compiler should put all global variables in the XSpec namespace if they are not in any namespace in the XSpec test as written; this would be more complicated to compile, because the compiler would also have to look for references to those variables and add the namespace prefix there.

Reply:
I think missing `@as` in `global-param` is just an oversight. Will be fixed in #350. (a222c74f9d58c9cab228cfceb10c433c7d6117f9 (fix) and bab2b036b8194914c3df2a92ebfc6c326f70aa90 (test), specifically)

Reply:
A private, dedicated namespace for test-specific variables sounds good. I'll plan to update files under `test\` that use `variable`, and add this to the list of proposed updates in the Compilation wiki. How about something like `xmlns:myv="http://example.org/ns/my/variable"`, to align with the existing use of `xmlns:my="http://example.org/ns/my"` in files like `xspec-variable.xspec`?

I am OK with using `param` instead of `variable` for global test-specific variables. I'll wait a little before making this change, just in case you change your mind or another reviewer chimes in.

Reply:
`xmlns:myv="http://example.org/ns/my/variable"` or any other will be fine, as long as there is a comment to clarify why the variable is namespaced.

As for the global variable, I'll make up my mind in due course. So let's keep it intact for now. (I need some time to look into the details anyway.) If we find no definite reason to decide what to do, maybe we can release the initial version without the global variable, wait for feedback and add it if demanded. Doing the other way round (i.e. supporting it at first and removing it afterwards) will be difficult.

Reply:
> A private, dedicated namespace for test-specific variables sounds good. I'll plan to update files under `test\` that use `variable`

Covered in 988163008627a4fa970b44253ec9cb70ec1ef139

Reply:
> Aside from the question of supported attributes, someone might choose to use param for overrides of globals defined in the file to be tested, reserving variable for variables that are defined and used only in the XSpec test. Do you think it's worthwhile to support that distinction in the mental model?

After all, now I think the global variable is worthwhile because
* In a few days ago, I looked at [a global `x:param`](https://github.com/xspec/xspec/blob/95f6b8cbd0f5d0638adf593d8dd9344c3fd9f98d/test/xspec-412.xspec#L4) which I had written myself, and wondered where in the tested XSLT the global parameter were defined. It took some minutes for me to realize that the `x:param` worked merely as a global variable.
* When explaining another basic concept, using `x:param` as a global variable (e.g. [this case](https://github.com/xspec/xspec/issues/132#issuecomment-469006666)) would just confuse the questioner who is rather new to XSpec.
* Most importantly, you've provided decent tests. 👍

Reply:
Sounds good to me. Commit b47edb84ad5992a2809aa54dce228a8ed2edf7e3 removes some code comments about this being an open question.